server/application/application.go

```diff
@@ -143,14 +143,13 @@ func (s *Server) List(ctx context.Context, q *application.ApplicationQuery) (*ap
 			newItems = append(newItems, *a)
 		}
 	}
-	if q.Name != nil {
```

Comment:
there is no need to check name is empty as FilterByName has handled this

Reply:
@crenshaw-dev inside application.pb.go it checks if nil and after FilterByName doing another check

```
func (m *ApplicationManifestQuery) GetName() string {
	if m != nil && m.Name != nil {
		return *m.Name
	}
	return ""
}
`````

```
func FilterByName(apps []argoappv1.Application, name string) ([]argoappv1.Application, error) {
	if name == "" {
		return apps, nil
	}
	
```

Reply:
But i actually agree with you, it is not mandatory to remove it in this PR